app/assets/javascripts/host_edit.js

```diff
@@ -363,11 +366,14 @@ function architecture_selected(element){
 }
 
 function os_selected(element){
-  var attrs = attribute_hash(['operatingsystem_id', 'organization_id', 'location_id']);
   var url = $(element).attr('data-url');
+  var type = $(element).attr('data-type');
+  var obj = (type == "hosts" ? "host" : "hostgroup");
+  var attrs = {};
+  attrs[obj] = attribute_hash(['operatingsystem_id', 'organization_id', 'location_id']);
```

Comment:
I tend to agree with Dominic in this case, the patch only changes the behavior and we will likely be backporting this into 1.13 RC2. On the other hand, if Katello is broken right now (RC1) and this patch changes it once again for RC2, it might be good idea to create such an extension.

Since this is important fix for 1.13, another PR could be good idea.